### ISSUE TITLE
Hoverable reciever chat text for other player || Sound on achievement recieve || WorldGuard achievements

### ISSUE THREAD

**Issue by AviRendall** [PyvesB/AdvancedAchievements#412]
It would be cool if there will be an option to enable/disable hoverable achievement recieve message for other player.
Example: https://pastebin.com/8VzdjEQ0

And option to play AWESOME_SOUND on achievement recieve. Not just *pling*.
Example: https://pastebin.com/tD4TxWbt

Also, i have an idea to add WorldGuard achievements.
Example: https://pastebin.com/yjmXHzFM

**Comment by PyvesB** [comment]
Hello there,

Maybe it makes sense to show the hoverable message when `ObfuscateNotYetReceived` is set to false? I'm not too sure, I'll have a think about it.

The problem with sounds it that the keys change on a regular basis in the API (there was a major overhaul in Minecraft 1.9, then again in Minecraft 1.13, with other smaller changes for most other Minecraft versions). I'm concerned about this and would like to avoid adding configuration options that may break on a regular basis.

As far as WorlGuard is concerned, quite a few users already have a similar setup. They simply configure WorlGuard to run _/aach give (`Commands` achievements) or _/aach add_ (`Custom` achievements) when a player enters or exits a region. There would probably not be that much benefit to add native support in Advanced Achievements. 😉 

Cheers,

Pyves

**Comment by PyvesB** [comment]
I've put some more thinking into the sound customisation. In the end I have added three new options, `SoundBook`, `SoundStats` and `SoundRanking`. As mentioned above, these may break when new Minecraft versions are released, but to minimise the impact on the users I've implemented proper fallback behaviour, so it's an acceptable compromise.

Here's a development version containing these new options. Let me know if you notice anything unexpected, other new features are shipped as well:
[AdvancedAchievements.zip](https://github.com/PyvesB/AdvancedAchievements/files/2425219/AdvancedAchievements.zip)

I have decided not to implement hoverable chat texts when other players receive achievements. The recommended main feature is the action bar notification, it makes little sense to add complexity to the configuration and pile more options on top of a alternate setting that is not widely used.
